yt/frontends/cf_radial/io.py

```diff
@@ -0,0 +1,67 @@
+"""
+CF-Radial-specific IO functions
+
+
+
+"""
+
+import numpy as np
+
+from yt.utilities.io_handler import BaseIOHandler
+
+
+class CFRadialIOHandler(BaseIOHandler):
+    _particle_reader = False
+    _dataset_type = "cf_radial"
+
+    def _read_particle_coords(self, chunks, ptf):
+        # This needs to *yield* a series of tuples of (ptype, (x, y, z)).
+        # chunks is a list of chunks, and ptf is a dict where the keys are
+        # ptypes and the values are lists of fields.
+        pass
+
+    def _read_particle_fields(self, chunks, ptf, selector):
+        # This gets called after the arrays have been allocated.  It needs to
+        # yield ((ptype, field), data) where data is the masked results of
+        # reading ptype, field and applying the selector to the data read in.
+        # Selector objects have a .select_points(x,y,z) that returns a mask, so
+        # you need to do your masking here.
+        pass
+
+    def _read_fluid_selection(self, chunks, selector, fields, size):
+        # This needs to allocate a set of arrays inside a dictionary, where the
+        # keys are the (ftype, fname) tuples and the values are arrays that
+        # have been masked using whatever selector method is appropriate.  The
+        # dict gets returned at the end and it should be flat, with selected
+        # data.  Note that if you're reading grid data, you might need to
+        # special-case a grid selector object.
+        # Also note that "chunks" is a generator for multiple chunks, each of
+        # which contains a list of grids. The returned numpy arrays should be
+        # in 64-bit float and contiguous along the z direction. Therefore, for
+        # a C-like input array with the dimension [x][y][z] or a
+        # Fortran-like input array with the dimension (z,y,x), a matrix
+        # transpose is required (e.g., using np_array.transpose() or
+        # np_array.swapaxes(0,2)).
+        chunks = list(chunks)
```

Comment:
I don't think this is necessary, and in fact it may have a negative impact on perfs

Reply:
ya, I'm not sure. These netcdf files will only have a single grid in a single chunk, so probably no performance issue in this case. We convert our chunk iterators to lists in many places throughout the code base, and it's not always clear to me why...

Reply:
in this particular case `chunks` is only used as an itertable, so I think the conversion isn't necessary.

Reply:
Ok, will test it out. I agree it's probably not necessary here...

Reply:
(ya, not necessary, took out the conversion)